server/forms.py

```diff
@@ -335,9 +336,19 @@ class TestJobForm(BaseForm):
     duration = IntegerField('Duration (seconds)', default=2)
 
 class MossSubmissionForm(BaseForm):
-    valid_languages = ['python', 'java', 'c', 'scheme', 'lisp', 'javascript']
     moss_userid = StringField('Your MOSS User ID',
                               validators=[validators.required()])
     file_regex = StringField('Regex for submitted files', default='.*',
                              validators=[validators.required()])
-    language = SelectField('Language', choices=[(pl, pl) for pl in valid_languages])
+    language = SelectField('Language', choices=[(pl, pl) for pl in COMMON_LANGUAGES])
+
+class GithubSearchRecentForm(BaseForm):
+    access_token = StringField('Github Access Token',
+                               description="Get a token at https://github.com/settings/tokens",
+                               validators=[validators.required()])
+    template_name = StringField('Template File Name',
+                                validators=[validators.required()])
+    keyword = StringField('Search for lines starting with', default="def ",
```

Comment:
I don't think it's very clear what this field is for. Does it extracts lines from the template before searching? What about leading white space (so we could e.g. match Java method names)